docs/kap_proposals/kap_7_remote_inventory.md

```diff
@@ -0,0 +1,64 @@
+# Remote Inventory Federation
+
+This feature would add the ability to Kapitan to fetch parts of the inventory from remote locations (https/git). This would allow users to combine different inventories from different sources and build modular infrastructure reusable across various repos.
+
+
+Author: @alpharoy14
+
+## Specification
+
+The configuration and declaration of remote inventories would be done in the hidden file `​.kapitan`.
+
+The file specifications are as follows:
+
+```yaml
+remoteInventory:
+    -   name: <inventory_name> #user defined
+        type: <inventory_type> #git\https
+        source: <source_of_inventory>
+        output_path: <relative_output_path>
+        ... type specific arguments
+```
+The command for adding remote inventories to the `.kapitan` file from cli will be:
+```$ kapitan inventory --add <inventory-name> <output-path> <source> ```
+The command for fetching remote inventories that is added to the `.kapitan` file will be:
+``` $ kapitan inventory --fetch <inventory-name> ```
+The command for fetching all the inventory declared in .kapitan file will be:
+
+```$ kapitan inventory --fetch-all```
+
+## Copying inventory files to the output location
+The output path is the path to save the inventory items into. The path is relative to the `inventory/` directory. For example, it could be `/classes/`. The contents of the fetched inventory will be recursively copied.
+
+The fetched inventory files will be cached in a temp directory.
+The temp directory (`remote_inventories/`) will be in the path pointed by the $XDG_CACHE_HOME environment variable. In case that does not exist, we can check the `$KAPITAN_CACHE_DIR`. Which, by default, will be set to the root folder of the project.
+
+## Force fetching
+While fetching, the output path will be recursively checked to see if it contains any file with the same name. If so, kapitan will raise an error saying so.
+
+To overwrite the files with the newly downloaded inventory items, we can add the --force flag to the fetch command, as shown below.
+
+`$ kapitan inventory --fetch-all --force`
+`$ kapitan inventory --fetch <inventory-name> --force`
```

Comment:
Just a thought: force fetching makes sense. But would this be the only option? Could we offer an interactive way of dealing with remote files (e.g. accept remote file instead, keep local file instead, etc...)

Reply:
I mentioned in the proposal that Kapitan would check if a file with the same name exists in the output path and would display a message to the user saying so. The user can then use the `--force` flag if they want to replace the existing file. Would you suggest that we have a boolean key in the configuration instead? 
eg 
```yaml
parameters:
  kapitan:
    inventory:
    - name: <inventory_name>
      type: <inventory_type>
      output_path: relative/path/to/file/or/dir
      source: <source_of_inventory>
      force: True
```
Or we can display a message (eg File already exists, would you like to force fetch it?) to which the user can respond with Y/N. Please let me know if there's a better way to make it more interactive.

Reply:
Just a thought really! I think we can park this for now and revisit when we have the main feature done.